src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3743,12 +3744,42 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                     if (!memberSignatures.TryGetValue(targetProperty, out var existingMember))
                     {
                         existingMember = OverriddenOrHiddenMembersHelpers.FindFirstHiddenMemberIfAny(targetProperty, memberIsFromSomeCompilation: true);
-                        isInherited = true;
+                        if (existingMember is FieldSymbol)
```

Comment:
>if (existingMember is FieldSymbol) [](start = 24, length = 34)

The logic feels too convoluted and I think it can be significantly simplified as follows:
1) First, we look for a matching property in the current record;
2) Then we look for a matching field in the current record;
3) Then we look for a first hidden member in base

Reply:
We shouldn't look at the base last though. A property from the base should win over a field from the current type, to avoid breaking existing scenario (below).
That said, I think we can achieve that by calling `FindFirstHiddenMemberIfAny` only once and keeping the result.

```
public record Base 
{
    public int X { get; set; } 
}
public record C(int X) : Base
{
    public new int X;
}
```

---
In reply to: [612815137](https://github.com/dotnet/roslyn/pull/52480#discussion_r612815137) [](ancestors = 612815137)

Reply:
> We shouldn't look at the base last though. A property from the base should win over a field from the current type, to avoid breaking existing scenario (below).

I don't think this behavior is sound, even though it avoids a breaking change.

---
In reply to: [612935164](https://github.com/dotnet/roslyn/pull/52480#discussion_r612935164) [](ancestors = 612935164,612815137)